msg.go

```diff
@@ -187,139 +187,167 @@ func packDomainName(s string, msg []byte, off int, compression map[string]int, c
 	if msg != nil {
 		lenmsg = len(msg)
 	}
+
 	ls := len(s)
 	if ls == 0 { // Ok, for instance when dealing with update RR without any rdata.
 		return off, 0, nil
 	}
-	// If not fully qualified, error out, but only if msg == nil #ugly
-	switch {
-	case msg == nil:
-		if s[ls-1] != '.' {
-			s += "."
-			ls++
-		}
-	case msg != nil:
-		if s[ls-1] != '.' {
+
+	// If not fully qualified, error out, but only if msg != nil #ugly
```

Comment:
why the fuck would `msg != nil` matter here? [Can't remember]

Reply:
I have no idea, I was just preserving existing semantics. I suspect it might have been from before `Len` was calculated separately from `Pack`. I can change it if you want?

Reply:
If you don't consider it part of the public API, I might be able to simplify this a lot and yank the `msg == nil` case out entirely. I don't think it's exercised anymore.